pkg/awsutils/awsutils.go

```diff
@@ -866,6 +866,11 @@ func (cache *EC2InstanceMetadataCache) AllocIPAddresses(eniID string, numIPs int
 		needIPs = ipLimit
 	}
 
+	// If we don't need any more IPs, exit
```

Comment:
864 has no check for err != nil.

Reply:
We should handle unknown instances better, but that's a different change.